### PR TITLE
perf(form-field): convert adapter to a class object

### PR DESCRIPTION
**Pull request by ngwattcos**
This fix converts the private adapter into a class object to hopefully improve memory usage

Edit: performance improvements should be comparable to #19980

### CODE REVIEW

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 1]
```diff
@@ -301,6 +321,7 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
               private _defaults?: MatFormFieldDefaultOptions,
               @Optional() @Inject(MAT_LABEL_GLOBAL_OPTIONS) private _labelOptions?: LabelOptions,
               @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
+    this._adapter = new TextFieldAdapter(this);
```
Comment (devversion) [inline src/material-experimental/mdc-form-field/form-field.ts:324]: could we set up the adapter above when we declare the `_adapter` class member?

Comment (ngwattcos) [reply src/material-experimental/mdc-form-field/form-field.ts]: @devversion Yep!

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 2]
```diff
@@ -91,6 +91,101 @@ const DEFAULT_FLOAT_LABEL: FloatLabelType = 'auto';
  */
 const FLOATING_LABEL_DEFAULT_DOCKED_TRANSFORM = `translateY(-50%)`;
 
+class TextFieldAdapter implements MDCTextFieldAdapter {
+  constructor(private _delegate: MatFormField) {}
+  addClass(className: string) {
+    this._delegate._textField.nativeElement.classList.add(className);
+  }
```
Comment (mmalerba) [inline src/material-experimental/mdc-form-field/form-field.ts:98]: newlines between methods

File: src/material-experimental/mdc-form-field/form-field.ts [hunk 3]
```diff
@@ -214,83 +322,7 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
...
-    registerTextFieldInteractionHandler: () => {},
-
-    // The foundation tries to setup a "MutationObserver" in order to watch for attributes
-    // like "maxlength" or "pattern" to change. The foundation will update the validity state
-    // based on that. We do not need this logic since we handle the validity through the
-    // abstract form control instance.
-    deregisterValidationAttributeChangeHandler: () => {},
-    registerValidationAttributeChangeHandler: () => null as any,
-  };
+  private _adapter: MDCTextFieldAdapter = new TextFieldAdapter(this);
```
Comment (kseamon) [inline src/material-experimental/mdc-form-field/form-field.ts:325]: As with other PRs, remove the _adaptor member and pass directly when creating the foundation

**Comment by jelbourn** [comment]
Blocked on https://github.com/material-components/material-components-web/pull/6256

**Comment by andrewseguin** [comment]
We will soon be taking out our dependency on the adapters provided by MDC since they won't be supported later. Thank you for your contribution, sorry we weren't able to land this

**Pull request closed without merging**